.github/copilot-instructions.md

```diff
@@ -0,0 +1,842 @@
+    MY_OBJECT* obj = THANDLE_GET_T(MY_OBJECT)(local_handle);
+    obj->data = 42;
+    
+    // Manual cleanup required - THANDLE does NOT have automatic scope cleanup
+    THANDLE_ASSIGN(MY_OBJECT)(&local_handle, NULL);  // Decrements ref count, frees if count reaches 0
+    
+    // Or let caller manage cleanup by not setting to NULL
+}
+```
+
+### Advanced THANDLE Patterns
+
+#### Function Parameters and Return Values
+```c
+// Passing THANDLE by value (preferred pattern)
+int process_object(THANDLE(MY_OBJECT) object)
+{
+    if (object == NULL)
+    {
+        return -1;
+    }
+    
+    // Use object->data directly
+    return object->data;
+    // No cleanup needed - caller retains ownership
+}
+
+// Returning THANDLE (caller gets ownership)
+THANDLE(MY_OBJECT) create_object_with_data(int initial_data)
+{
+    THANDLE(MY_OBJECT) result = THANDLE_MALLOC(MY_OBJECT)(dispose_function);
+    if (result != NULL)
+    {
+        THANDLE_GET_T(MY_OBJECT)(result)->data = initial_data;
+    }
+    return result;  // Caller gets ownership
+}
+
+// Output parameter pattern (for optional results)
+int try_get_object(THANDLE(MY_OBJECT)* out_object)
+{
+    int result;
+    THANDLE(MY_OBJECT) temp = create_some_object();
+    if (temp == NULL)
+    {
+        result = MU_FAILURE;
+    }
+    else
+    {
+        THANDLE_MOVE(MY_OBJECT)(out_object, &temp);  // Transfer to caller
+        result = 0;
+    }
+    
+    return result;
+}
+```
+
+#### Thread Safety and Shared Objects
+```c
+// THANDLE provides thread-safe reference counting
+// Multiple threads can safely share THANDLE instances
+
+// Worker thread function
+int worker_thread(void* param)
+{
+    THANDLE(MY_OBJECT) obj = param;
+    
+    // Use the shared object
+    MY_OBJECT* data = THANDLE_GET_T(MY_OBJECT)(obj);
+    process_data(data);
+    
+    // Clean up by assigning to NULL
+    THANDLE_ASSIGN(MY_OBJECT)(&obj, NULL);
+    return 0;
+}
+
+// Example: Create THANDLE and pass to thread
+void example_thread_usage(void)
+{
+    // Create THANDLE
+    THANDLE(MY_OBJECT) shared_obj = THANDLE_MALLOC(MY_OBJECT)(dispose_function);
+    if (shared_obj == NULL)
+    {
+        LogError("Failed to create THANDLE object");
+        return;
+    }
+    
+    // Assign to variable for thread
+    THANDLE(MY_OBJECT) thread_obj = NULL;
+    THANDLE_ASSIGN(MY_OBJECT)(&thread_obj, shared_obj);
+    
+    // Pass to thread
+    THREAD_HANDLE thread;
+    if (ThreadAPI_Create(&thread, worker_thread, thread_obj) != THREADAPI_OK)
+    {
+        LogError("Failed to create thread");
+        THANDLE_ASSIGN(MY_OBJECT)(&thread_obj, NULL);  // Clean up thread reference
+    }
+    else
+    {
+        // Wait for thread to complete
+        if (ThreadAPI_Join(thread, NULL) != THREADAPI_OK)
+        {
+            LogError("Failed to join thread");
+        }
+    }
+    
+    // Clean up original reference
+    THANDLE_ASSIGN(MY_OBJECT)(&shared_obj, NULL);
+}
+```
+
+### THANDLE vs Raw Pointers vs REFCOUNT
+
```

Comment:
Table has an inconsistent column ordering:
Chapter name is 1-2-3
Table is 2-3-1